src/backend/utils/adt/varlena.c

```diff
@@ -42,6 +42,9 @@
 #include "utils/varlena.h"
 
 
+pltsql_strpos_non_determinstic_hook_type pltsql_strpos_non_determinstic_hook = NULL;
```

Comment:
Is it possible to introduce sys version of these functions?

Reply:
Used hooks here so that we do not change anything for deterministic collation, i.e. existing behaviour.

Writing a sys function for this might be tricky because we have three levels of handling in babelfish.
1. deterministic CS_AS
2. non deterministic CI_AS (converted to deterministic internally)
3. non deterministic CI_AI (implemented using hooks)